edx_django_utils/management/commands/manage_user.py

```diff
@@ -0,0 +1,183 @@
+"""
+Management command `manage_user` is used to idempotently create or remove
+Django users, set/unset permission bits, and associate groups by name.
+"""
+
+
+import random
+import string
+
+import django.dispatch
+from django.contrib.auth import get_user_model
+from django.contrib.auth.hashers import identify_hasher, is_password_usable
+from django.contrib.auth.models import Group
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.utils.translation import gettext as _
+
+manage_user_cmd = django.dispatch.Signal(providing_args=["user"])
+
+
+def is_valid_django_hash(encoded):
+    """
+    Starting with django 2.1, the function is_password_usable no longer checks whether encode
+    is a valid password created by a django hasher (hasher in  PASSWORD_HASHERS setting)
+
+    Adding this function to create constant behavior as we upgrade django versions
+    """
+    try:
+        identify_hasher(encoded)
+    except ValueError:
+        return False
+    return True
+
+
+def generate_password(length=12, chars=string.ascii_letters + string.digits):
+    """Generate a valid random password"""
+    if length < 8:
+        raise ValueError("password must be at least 8 characters")
+
+    choice = random.SystemRandom().choice
+
+    password = ''
+    password += choice(string.digits)
+    password += choice(string.ascii_letters)
+    password += ''.join([choice(chars) for _i in range(length - 2)])
+    return password
+
+
+class Command(BaseCommand):  # lint-amnesty, pylint: disable=missing-class-docstring
```

Comment:
@kdmccormick these tests are written specifically for edx-platform. We expect to use the management commands with other IDAs for granting permissions.

I am not sure how will the tests run if placed in this repo.

Reply:
Hmm, it looks like this repo may not have a ton of test coverage and think it may be reasonable to add a couple unit tests for the sub functions into https://github.com/edx/edx-django-utils/tree/master/edx_django_utils/tests

I am not sure how to test the handle function without doing a lot of work though.

@kdmccormick do you think it would be reasonable to leave these two tests in edx-platform and then add new tests to the repo to increase test coverage?

Reply:
@adzuci @NIXKnight I could be wrong, but `test_manage_user.py` and `test_manage_group.py` from edx-platform seem like they would work fine if copied over to edx-django-utils with a couple tweaks to remove the `user.profile` checks. That strikes me as less work than writing brand new tests for this repo.

The original tests in edx-platform could be left as-is, or cut down so that they are only testing the edx-platform-specific `user.profile` aspect.